src/pkg/clients/Service_discovery.go

```diff
@@ -36,49 +36,70 @@ func NewServiceDiscovery(host, user, password string) (ServiceDiscovery, error)
 		return nil, err
 	}
 
-	// Create a Consul agent client
-	agent := client.Agent()
-	service := serviceDiscovery{agent: agent}
-
-	return service, nil
-}
-
-// Get all available instances of transformation services
-func (s serviceDiscovery) GetTransformationServices() ([]*TransformerInfos, error) {
-	services, err := s.agent.ServicesWithFilter("transformer in Service")
-	if err != nil {
-		return nil, err
+	// Create service discovery
+	service := serviceDiscovery{
+		client:                    client,
+		agent:                     client.Agent(),
+		transformersAddressesList: map[string][]string{},
 	}
-	return s.parseTransformerInfos(services), nil
-}
 
-// Get all available instances of a given transformation service
-func (s serviceDiscovery) GetTransformationServicesWithName(name string) ([]*TransformerInfos, error) {
-	services, err := s.agent.ServicesWithFilter(name + " in Service and transformer in Service")
-	if err != nil {
-		return nil, err
-	}
-	return s.parseTransformerInfos(services), nil
+	return &service, nil
 }
 
-func (s serviceDiscovery) parseTransformerInfos(services map[string]*consul_api.AgentService) []*TransformerInfos {
-	transformers := make([]*TransformerInfos, 0, len(services))
-	for _, service := range services {
-		transformers = append(transformers, &TransformerInfos{
-			Name:    service.Service,
-			Address: service.Address,
-			Port:    strconv.Itoa(service.Port),
-		})
+// Get all available instances of a given transformation service
+func (s *serviceDiscovery) GetTransformationServices(name string) ([]string, error) {
+	if len(s.transformersAddressesList[name]) < 1 {
+		return nil, fmt.Errorf("No service with name %v found.", name)
 	}
-	return transformers
+	return s.transformersAddressesList[name], nil
 }
 
 // Register service, will be use only for local dev
-func (s serviceDiscovery) RegisterService(name, address string, port int, tags []string) error {
+func (s *serviceDiscovery) RegisterService(name, address string, port int, tags []string) error {
 	return s.agent.ServiceRegister(&consul_api.AgentServiceRegistration{
 		Name:    name,
 		Address: address,
 		Port:    port,
 		Tags:    tags,
 	})
 }
+
+// Get all available instances of transformation services
+func (s *serviceDiscovery) updateList() error {
+	services, err := s.agent.ServicesWithFilter("transformer in Service")
+	if err != nil {
+		return err
+	}
+	s.parseTransformerList(services)
+	return nil
+}
+
+func (s *serviceDiscovery) parseTransformerList(services map[string]*consul_api.AgentService) {
+	s.transformersAddressesList = map[string][]string{}
+	for _, service := range services {
+		name := strings.Split(service.Service, "-")[0]
+		address := service.Address + ":" + strconv.Itoa(service.Port)
+		s.transformersAddressesList[name] = append(s.transformersAddressesList[name], address)
+	}
+}
+
+func (s *serviceDiscovery) Watch() error {
```

Comment:
Here we should probably have someting like
```suggestion
func (s *serviceDiscovery) Watch(ctx context.Context) error {
```

and some listener like

```golang
	go func {
            for {
		select {
		case <-ctx.Done():
			fmt.Printf("gracefully shutdown consul watcher\n")
			plan.Stop()
			return
		}
	    }
        }
```

And test it manually ! (by CTRL+C typically)